apps/Gruntfile.js

```diff
@@ -706,22 +706,99 @@ describe('entry tests', () => {
           qtip2: 'var $'
         }
       ],
+      mode: minify ? 'production' : 'development',
+      optimization: {
+        minimizer: [
+          compiler => {
+            const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+            const plugin = new UglifyJsPlugin({
+              cache: true,
+              parallel: true,
+              sourceMap: envConstants.DEBUG_MINIFIED
+            });
+            plugin.apply(compiler);
+          }
+        ],
+
+        // We use a single, named runtimeChunk in order to be able to load
+        // multiple webpack entry points on a single page. **The resulting
+        // 'webpack-runtime' chunk must be included exactly once on each page
+        // which includes webpack entry points.** If you do not include the
+        // runtime, webpack entry points you include will not be loaded. If you
+        // include the runtime twice, webpack entry points will be loaded twice.
+        //
+        // Without a single, named runtimeChunk there would be no runtimeChunk
+        // to include, and entry points would load and run separately.
+        // However, those entry points would create separate instances of any
+        // shared modules. This would mean that state within webpack modules
+        // cannot be shared between entry points, breaking many assumptions made
+        // by our application. For more information, see:
+        // https://webpack.js.org/concepts/manifest/#runtime
+        // https://webpack.js.org/configuration/optimization/#optimizationruntimechunk
+        //
+        // In the future, if we can limit ourselves to one webpack entry point
+        // per page, we could consider removing the runtimeChunk config.
+        runtimeChunk: {
+          name: 'webpack-runtime'
+        },
+        splitChunks: {
+          cacheGroups: {
+            // Pull any module shared by 2+ appsEntries into the "common" chunk.
+            common: {
+              name: 'common',
+              minChunks: 2,
+              chunks: chunk => {
+                return _.keys(appsEntries).includes(chunk.name);
+              }
+            },
+            // Pull any module shared by 2+ codeStudioEntries into the
+            // "code-studio-common" chunk.
+            'code-studio-common': {
+              name: 'code-studio-common',
+              minChunks: 2,
+              chunks: chunk => {
+                const chunkNames = _.keys(codeStudioEntries);
+                return chunkNames.includes(chunk.name);
+              },
+              priority: 10
```

Comment:
What does this setting do?

Reply:
https://webpack.js.org/plugins/split-chunks-plugin/#splitchunkscachegroupspriority

When multiple cacheGroups want to extract the same module from the same entry point, SplitChunksPlugin will use the cacheGroup with the higher priority to decide which cacheGroup to use. 

In this case, the only important thing to note about the priorities is that `code-studio-multi` has higher priority than `common`, so that's why anything that `code-studio-multi` matches will "pull things out of" `common`.